coriolis_openstack_utils/actions.py

```diff
@@ -1090,3 +1091,143 @@ def execute_operations(self):
+
+    def __init__(self, source_client, destination_client, action_payload):
+        """
+        param action_payload: dict(): payload (params) for the action
+        must contain 'src_network_id'
+                     'dest_tenant_id'
+        """
+        # NOTE no openstack_client or coriolis_client
+
+        super(NetworkCreationAction, self).__init__(
+            None, None, action_payload)
+        self._source_client = source_client
+        self._destination_client = destination_client
+
+    @property
+    def network_name_format(self):
+        return CONF.destination.new_network_name_format
+
+    def check_already_done(self):
+        src_network = networks.get_network(
+            self._source_client, self.payload['src_network_id'])
+
+        dest_network_name = self.get_new_network_name()
+        conflicting = networks.list_networks(
+            self._destination_client, self.payload['dest_tenant_id'],
+            filters={'name': dest_network_name})
+
+        for dest_network in conflicting:
+            if networks.check_network_similarity(
+                    src_network, dest_network,
+                    self._source_client, self._destination_client):
+                LOG.info("Found destination network '%s' with same "
+                         "information and subnets as source network '%s'."
+                         % (dest_network_name, src_network['name']))
+                return {"done": True, "result": dest_network['id']}
+
+        if len(conflicting) == 1:
+            raise Exception("Found destination network with "
+                            "with name '%s' but different attributes!"
+                            % dest_network_name)
+        elif conflicting:
+            raise Exception("Found multiple destination networks with "
+                            "with name '%s'! Aborting subnet "
+                            "migration." % dest_network_name)
+
+        return {"done": False, "result": None}
+
+    def equivalent_to(self, other_action):
+        if other_action.action_type == self.action_type:
+            if self.payload['src_network_id'] == (
+                    other_action.payload.get('src_network_id')):
+                return (self.payload.get('dest_tenant_id') ==
+                        other_action.payload.get('dest_tenant_id'))
+        return False
+
+    def print_operations(self):
+        super(NetworkCreationAction, self).print_operations()
+        network_name = self.get_new_network_name()
+        LOG.info(
+            "Create new destination network named '%s'." % network_name)
+
+    def get_source_network_name(self):
+        return networks.get_network(
+            self._source_client, self.payload['src_network_id'])['name']
+
+    def get_new_network_name(self):
+        return self.network_name_format % {
+            "original": self.get_source_network_name()}
+
+    def create_network_body(self, description):
```

Comment:
Considering you decided to put "supporting operations" in dedicated `{networks,subnets}.py` modules (which is a good call), I think it would be better to have "extra bits of logic" in the separate files and only keep base interface method implementations in the Action classes (of course you can leave "trivial accessor-like methods" such as `get_new_network_name()` here (that'll also come in handy when we do the whole-tenant migration thing))